opm/simulators/linalg/FlowLinearSolverParameters.hpp

```diff
@@ -255,21 +257,23 @@ namespace Opm
             // TODO: these parameters have undocumented non-trivial dependencies
             linear_solver_reduction_ = EWOMS_GET_PARAM(TypeTag, double, LinearSolverReduction);
             relaxed_linear_solver_reduction_ = EWOMS_GET_PARAM(TypeTag, double, RelaxedLinearSolverReduction);
-            ilu_relaxation_ = EWOMS_GET_PARAM(TypeTag, double, IluRelaxation);
             linear_solver_maxiter_ = EWOMS_GET_PARAM(TypeTag, int, LinearSolverMaxIter);
             linear_solver_restart_ = EWOMS_GET_PARAM(TypeTag, int, LinearSolverRestart);
             linear_solver_verbosity_ = EWOMS_GET_PARAM(TypeTag, int, LinearSolverVerbosity);
+            ilu_relaxation_ = EWOMS_GET_PARAM(TypeTag, double, IluRelaxation);
             ilu_fillin_level_ = EWOMS_GET_PARAM(TypeTag, int, IluFillinLevel);
             ilu_milu_ = convertString2Milu(EWOMS_GET_PARAM(TypeTag, std::string, MiluVariant));
             ilu_redblack_ = EWOMS_GET_PARAM(TypeTag, bool, IluRedblack);
             ilu_reorder_sphere_ = EWOMS_GET_PARAM(TypeTag, bool, IluReorderSpheres);
             newton_use_gmres_ = EWOMS_GET_PARAM(TypeTag, bool, UseGmres);
             ignoreConvergenceFailure_ = EWOMS_GET_PARAM(TypeTag, bool, LinearSolverIgnoreConvergenceFailure);
             scale_linear_system_ = EWOMS_GET_PARAM(TypeTag, bool, ScaleLinearSystem);
+            linsolver_ = EWOMS_GET_PARAM(TypeTag, std::string, LinearSolver);
+            linear_solver_print_json_definition_ = EWOMS_GET_PARAM(TypeTag, bool, LinearSolverPrintJsonDefinition);
             cpr_reuse_setup_  =  EWOMS_GET_PARAM(TypeTag, int, CprReuseSetup);
             cpr_reuse_interval_  =  EWOMS_GET_PARAM(TypeTag, int, CprReuseInterval);
 
-            if (!EWOMS_PARAM_IS_SET(TypeTag, int, LinearSolverMaxIter) && cprRequestedInDataFile) {
+            if (!EWOMS_PARAM_IS_SET(TypeTag, int, LinearSolver) && cprRequestedInDataFile) {
```

Comment:
This is a change I added just now that is not just a cleanup, but in my mind a bugfix: If the data file contains CPR we shall use the "cpr" preconditioner unless the command line overrides it. But the command line argument that overrides it is not LinearSolverMaxIter but LinearSolver.